src/parser.rs

```diff
@@ -1159,6 +1159,32 @@ impl<'a> Parser<'a> {
                     }
                 }
                 Keyword::XOR => Some(BinaryOperator::Xor),
+                Keyword::OPERATOR if dialect_of!(self is PostgreSqlDialect) => {
```

Comment:
Given @ovr 's comment here, perhaps we should also add support when parsing in `GenericDialect`? https://github.com/apache/arrow-datafusion/pull/3037#issuecomment-1205609941

Reply:
Done! I kept the name as `PGCustomOperator` but happy to change it as well.